core/js/src/test/scala/com/softwaremill/sttp/testing/HttpTestExtensions.scala

```diff
@@ -1,24 +1,66 @@
 package com.softwaremill.sttp.testing
 
+import scala.concurrent.ExecutionContext
 import scala.concurrent.Future
+import scala.concurrent.Promise
 import scala.language.higherKinds
 import scala.scalajs.js
 import scala.scalajs.js.JSConverters._
+import scala.scalajs.js.JavaScriptException
+import scala.scalajs.js.annotation.JSGlobal
+import scala.scalajs.js.typedarray.AB2TA
 
 import com.softwaremill.sttp.dom.experimental.FilePropertyBag
 import com.softwaremill.sttp.dom.experimental.{File => DomFile}
 import com.softwaremill.sttp.internal.SttpFile
+import org.scalajs.dom.FileReader
+import org.scalajs.dom.raw.Event
+import org.scalajs.dom.raw.UIEvent
 
 trait HttpTestExtensions[R[_]] { self: HttpTest[R] =>
 
-  protected def withTemporaryFile[T](content: String)(f: SttpFile => Future[T]): Future[T] = {
+  implicit override def executionContext: ExecutionContext = ExecutionContext.Implicits.global
```

Comment:
isn't this always `global`? If so, maybe it can be specified in the base trait, or just through an import?

Reply:
The default ScalaTest execution context uses the Scala.js Queue execution context which errors when running async tests. I'll pull it out into a trait so it is obvious why it is needed.